x-pack/plugins/security_solution/public/timelines/components/timeline/kpi/kpi_container.tsx

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+import { isEmpty, pick } from 'lodash/fp';
+import { useSelector } from 'react-redux';
+import { getEsQueryConfig } from '@kbn/data-plugin/common';
+import type { TimerangeInput } from '@kbn/timelines-plugin/common';
+import { EuiPanel } from '@elastic/eui';
+import { TimelineId } from '../../../../../common/types';
+import { useSourcererDataView } from '../../../../common/containers/sourcerer';
+import type { State } from '../../../../common/store';
+import { useDeepEqualSelector } from '../../../../common/hooks/use_selector';
+import { SourcererScopeName } from '../../../../common/store/sourcerer/model';
+import { TimelineKPIs } from './kpis';
+import { useTimelineKpis } from '../../../containers/kpis';
+import { useKibana } from '../../../../common/lib/kibana';
+import { timelineSelectors } from '../../../store/timeline';
+import { timelineDefaults } from '../../../store/timeline/defaults';
+import { combineQueries } from '../../../../common/lib/kuery';
+import {
+  endSelector,
+  startSelector,
+} from '../../../../common/components/super_date_picker/selectors';
+
+interface KpiExpandedProps {
+  timelineId: string;
+}
+
+export const TimelineKpisContainer = ({ timelineId }: KpiExpandedProps) => {
+  const { browserFields, indexPattern, selectedPatterns } = useSourcererDataView(
+    SourcererScopeName.timeline
+  );
+
+  const { uiSettings } = useKibana().services;
+  const esQueryConfig = useMemo(() => getEsQueryConfig(uiSettings), [uiSettings]);
+  const getTimeline = useMemo(() => timelineSelectors.getTimelineByIdSelector(), []);
+  const { dataProviders, filters, kqlMode } = useDeepEqualSelector((state) =>
+    pick(
+      [
+        'activeTab',
+        'dataProviders',
+        'kqlQuery',
+        'status',
+        'title',
+        'timelineType',
+        'updated',
+        'show',
+        'filters',
+        'kqlMode',
+      ],
+      getTimeline(state, timelineId) ?? timelineDefaults
+    )
+  );
+
+  const getKqlQueryTimeline = useMemo(() => timelineSelectors.getKqlFilterQuerySelector(), []);
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const kqlQueryTimeline = useSelector((state: State) => getKqlQueryTimeline(state, timelineId)!);
+
+  const kqlQueryExpression =
+    isEmpty(dataProviders) && isEmpty(kqlQueryTimeline) ? ' ' : kqlQueryTimeline;
```

Comment:
Do we really need the _no-non-null-assertion_ exception here? `isEmpty(undefined)` will return `true` as for `''`.
Which makes me think that this probably should be an _OR_ condition, right? 
`const kqlQueryExpression = isEmpty(dataProviders) || isEmpty(kqlQueryTimeline) ? ' ' : kqlQueryTimeline;`

Reply:
I think `isEmpty(dataProviders)` shouldn't event be here.. since its value does not impact either `kqlQueryTimeline` or `kqlQueryExpression` so I have put below as the final condition.

`isEmpty(kqlQueryTimeline) ? ' ' : kqlQueryTimeline;`

Reply:
Sounds good, then you should be able to remove the `no-non-null-assertion` exception, right?

Reply:
Thanks for addressing it